File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/GeoCode.java

package info.mx.comlib.retrofit.service.model.google;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class GeoCode implements Serializable, Parcelable {

    @SerializedName("results")
    @Expose
    private List<Result> results = null;
    @SerializedName("status")
    @Expose
    private String status;
    public final static Creator<GeoCode> CREATOR = new Creator<GeoCode>() {


        @SuppressWarnings({
                "unchecked"
        })
        public GeoCode createFromParcel(Parcel in) {
            return new GeoCode(in);
        }

        public GeoCode[] newArray(int size) {
            return (new GeoCode[size]);
        }

    };
    private final static long serialVersionUID = 3171710963124198211L;

    protected GeoCode(Parcel in) {
        in.readList(this.results, (Result.class.getClassLoader()));
        this.status = ((String) in.readValue((String.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     */
    public GeoCode() {
    }

    /**
     * @param results
     * @param status
     */
    public GeoCode(List<Result> results, String status) {
        super();
        this.results = results;
        this.status = status;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeList(results);
        dest.writeValue(status);
    }

    public int describeContents() {
        return 0;
    }

}

File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/GeoCode.kt
package info.mx.comlib.retrofit.service.model.google

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serial
import java.io.Serializable

open class GeoCode : Serializable, Parcelable {
    @SerializedName("results")
    @Expose
    var results: MutableList<Result?>? = null

    @SerializedName("status")
    @Expose
    var status: String? = null

    protected constructor(`in`: Parcel) {
        `in`.readList(this.results!!, (Result::class.java.classLoader))
        this.status = (`in`.readValue((String::class.java.classLoader)) as String?)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeList(results)
        dest.writeValue(status)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<GeoCode?> = object : Parcelable.Creator<GeoCode?> {
            override fun createFromParcel(`in`: Parcel): GeoCode {
                return GeoCode(`in`)
            }

            override fun newArray(size: Int): Array<GeoCode?> {
                return (arrayOfNulls<GeoCode>(size))
            }
        }

        @Serial
        private const val serialVersionUID = 3171710963124198211L
    }
}
